packages/block-editor/src/components/link-control/index.js

```diff
@@ -158,6 +158,7 @@ const makeCancelable = ( promise ) => {
 function LinkControl( {
 	searchInputPlaceholder,
 	value,
+	id,
```

Comment:
I'm not sure about introducing `id` as a separate prop. It's really just introducing a second `value`, but there's already a `value` prop that should serve this purpose.

It feels like instead the Navigation Link block should be able to omit the `url` from the `value` prop, instead defining the `id` to achieve the same thing:
```js
{
    id: 1
}
```

and for custom links it could omit the `id`, instead specifying the `url`:
```js
{
    url: 'https://google.com',
}
```